srml/staking/src/lib.rs

```diff
@@ -615,9 +607,22 @@ decl_storage! {
 		/// A mapping from still-bonded eras to the first session index of that era.
 		BondedEras: Vec<(EraIndex, SessionIndex)>;
 
-		/// All slashes that have occurred in a given era.
-		EraSlashJournal get(era_slash_journal):
-			map EraIndex => Vec<SlashJournalEntry<T::AccountId, BalanceOf<T>>>;
+		/// All slashing events on validators, mapped by era to the highest slash proportion
+		/// and slash value of the era.
+		ValidatorSlashInEra:
+			double_map EraIndex, twox_128(T::AccountId) => Option<(Perbill, BalanceOf<T>)>;
+
+		/// All slashing events on nominators, mapped by era to the highest slash value of the era.
+		NominatorSlashInEra:
+			double_map EraIndex, twox_128(T::AccountId) => Option<BalanceOf<T>>;
+
+		/// Slashing spans for stash accounts.
+		SlashingSpans: map T::AccountId => Option<slashing::SlashingSpans>;
+
+		/// Records information about the maximum slash of a stash within a slashing span,
+		/// as well as how much reward has been paid out.
+		SpanSlash: map (T::AccountId, slashing::SpanIndex)
```

Comment:
this is something that's fairly likely to change in the near future. what's the typical approach in SRML for making storage entries versionable?

Reply:
oh, if it is deployed then it needs manual, one-time  migration code `on_initialised`.

Reply:
This https://github.com/paritytech/substrate/pull/3947/files should make it at least less of an annoyance.